cmd/testmatrix/main.go

```diff
@@ -0,0 +1,237 @@
+				</td>
+				{{- end }}
+				<td>{{ $platform }}</td>
+				{{- range $scenario := $.Scenarios }}
+					<td>{{ index $.RepoToPlatformToScenario $repoName $platform $scenario }}</td>
+				{{- end }}
+			</tr>
+		{{- end }}
+	{{- end }}
+	</tbody>
+</table>
+
+- *{{ .Pass }} means passing*
+- *{{ .Skip }} means not implemented (skipped)*
+`
+)
+
+type Args struct {
+	e2e_testing.CmdWithProjectId
+}
+
+type status string
+
+type result struct {
+	RepoName string
+	Platform string
+	Statuses map[string]status
+}
+
+var (
+	triggerNameRe  = regexp.MustCompile(`^ops-\w+-e2e-.*$`)
+	scenarioPassRe = regexp.MustCompile(`: --- PASS:\s+([\w_]+)`)
+	scenarioSkipRe = regexp.MustCompile(`: --- SKIP:\s+([\w_]+)`)
+)
+
+func main() {
+	args := Args{}
+	arg.MustParse(&args)
+
+	ctx := context.Background()
+	cloudbuildService, err := cloudbuild.NewService(ctx)
+	if err != nil {
+		panic(err)
+	}
+	storageClient, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	listTriggersRes, err := cloudbuildService.Projects.Triggers.List(args.ProjectID).
+		Context(ctx).
+		PageSize(128).
+		Do()
+	if err != nil {
+		panic(err)
+	}
+
+	var wg sync.WaitGroup
+	results := make([]*result, len(listTriggersRes.Triggers))
+	for i, trigger := range listTriggersRes.Triggers {
+		wg.Add(1)
+		go handleTrigger(
+			ctx,
+			&wg,
+			args.ProjectID,
+			&results[i],
+			trigger,
+			cloudbuildService,
+			storageClient,
+		)
+	}
+	wg.Wait()
+
+	repoToPlatformToScenario := map[string]map[string]map[string]status{}
+	repoNameSet := map[string]struct{}{}
+	scenarioSet := map[string]struct{}{}
+	platformSet := map[string]struct{}{}
+	for _, result := range results {
+		if result == nil || result.Platform == "build" {
+			continue
+		}
+
+		repoNameSet[result.RepoName] = struct{}{}
+		if repoToPlatformToScenario[result.RepoName] == nil {
+			repoToPlatformToScenario[result.RepoName] = map[string]map[string]status{}
+		}
+		platformToScenario := repoToPlatformToScenario[result.RepoName]
+
+		platformToScenario[result.Platform] = result.Statuses
+		platformSet[result.Platform] = struct{}{}
+
+		for scenario := range result.Statuses {
+			scenarioSet[scenario] = struct{}{}
+		}
+	}
+	repoNames := sortStringSet(repoNameSet)
+	scenarios := sortStringSet(scenarioSet)
+	platforms := sortStringSet(platformSet)
+
+	template := template.Must(template.New("table").Parse(templateTxt))
+	err = template.Execute(os.Stdout, struct {
+		RepoNames                []string
+		Scenarios                []string
+		Platforms                []string
+		RepoToPlatformToScenario map[string]map[string]map[string]status
+		Pass                     status
+		Skip                     status
+	}{repoNames, scenarios, platforms, repoToPlatformToScenario, pass, skip})
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+// handleTrigger populates the resPtr with the results of the latest build of
+// the given trigger.
+func handleTrigger(
+	ctx context.Context,
+	wg *sync.WaitGroup,
```

Comment:
See above: it's less error prone to remove the wg argument and have the caller handle synchronization.